### PR TITLE
Convert groups class to hook

### PR DESCRIPTION
**Pull request by fhlavac**
- replaced `connect()` functions with `useSelector()` and `useDispatch()`
- replaced history prop in groups with `useHistory()`
- removed redundant code
- updated tests and enhanced with pagination and filtering tests

### CODE REVIEW

File: src/smart-components/group/groups.js [hunk 1]
```diff
@@ -1,49 +1,65 @@
...
   const routes = () => <Fragment>
-    <Route exact path="/groups/add-group" render={ props => <AddGroupWizard { ...props } postMethod={ fetchData } /> } />
-    <Route exact path="/groups/edit/:id" render={ props => <EditGroup { ...props } postMethod={ fetchData } isOpen/> } />
-    <Route exact path="/groups/removegroups" render={ props => <RemoveGroup { ...props } postMethod={ (ids) => {
-      fetchData();
-      setSelectedRows(selectedRows.filter(row => (!ids.includes(row.uuid))));
-      setFilterValue('');
-    } } isModalOpen groupsUuid={ removeGroupsList } /> } />
+    <Route exact path="/groups/add-group"
+      render={ props => <AddGroupWizard { ...props } postMethod={ config => dispatch(fetchGroups(config)) } /> }
```
Comment (Hyperkid123) [inline src/smart-components/group/groups.js]: While you are doing this, you should use the children syntax
```jsx
<Route>
  <AddGroupWizard postMethod={ config => dispatch(fetchGroups(config)) } />
</Route>
```
The `{...props}` should be replaced inside the component with hooks like `useHistory`, `useRouteMatch`, etc.

File: src/smart-components/group/groups.js [hunk 2]
```diff
@@ -1,49 +1,65 @@
...
 const columns = [{ title: 'Name', cellFormatters: [ expandable ]}, 'Description', 'Members', 'Last modified' ];
 
-const Groups = ({ fetchGroups, isLoading, pagination, history: { push }, groups, userIdentity }) => {
+const Groups = () => {
   const [ filterValue, setFilterValue ] = useState('');
   const [ selectedRows, setSelectedRows ] = useState([]);
   const [ removeGroupsList, setRemoveGroupsList ] = useState([]);
 
+  const dispatch = useDispatch();
+  const { groups, pagination, userIdentity, isLoading } = useSelector(({ groupReducer: { groups, isLoading }}) => ({
```
Comment (karelhala) [inline src/smart-components/group/groups.js]: Can you add default value here as well? So it doesn't break for no reason.
```JS
const { groups, pagination, userIdentity, isLoading } = useSelector((
  { groupReducer: { groups, isLoading }} = { groupReducer: {} }
) => ({
```

Comment (Hyperkid123) [reply src/smart-components/group/groups.js]: I would solve this on the groupReducer level. There should always be an initial state.

Comment (fhlavac) [reply src/smart-components/group/groups.js]: Currently I have
```
// Initial State
export const groupsInitialState = {
  groups: {
    data: [],
    meta: {
      count: 0,
      limit: 10,
      offset: 0
    }
  },
  selectedGroup: { addRoles: {}},
  isLoading: false,
  isRecordLoading: false
};
```
in group-reducer.js, so it should not fail, should it? There are default values for everything.

Comment (Hyperkid123) [reply src/smart-components/group/groups.js]: Then there should be no problem

File: src/smart-components/group/add-group/add-group-wizard.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import React, { useState } from 'react';
-import { withRouter } from 'react-router-dom';
+import { withRouter, useHistory } from 'react-router-dom';
```
Comment (Hyperkid123) [inline src/smart-components/group/add-group/add-group-wizard.js]: Is anything from with router still used?

File: src/smart-components/group/remove-group-modal.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
-import { withRouter } from 'react-router-dom';
+import { withRouter, useHistory } from 'react-router-dom';
```
Comment (karelhala) [inline src/smart-components/group/remove-group-modal.js]: Same as with `add-group-wizard.js` this component no longer uses anything from decorator `withRouter`.

**Pull request merged**